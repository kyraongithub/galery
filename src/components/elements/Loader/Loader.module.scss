.loader {
  position: relative;
  margin: auto;
  width: 12px;
  height: 12px;
  border: 3px solid #000;
  border-top: 3px solid #fff !important;
  border-radius: 50%;
  animation: spin 2s linear infinite;
}

.big {
  width: 48px;
  height: 48px;
}

.white {
  border: 3px solid #fff;
  border-top: 3px solid #000 !important;
}

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
