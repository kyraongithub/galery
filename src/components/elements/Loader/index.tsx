import cn from "classnames";
import styles from "./Loader.module.scss";
import { Props } from "./type.type";

const Loader = (props: Props) => {
  const { big, white } = props;

  return (
    <div
      className={cn(
        styles.loader,
        { [styles.big]: big },
        { [styles.white]: white }
      )}
    ></div>
  );
};

export default Loader;
