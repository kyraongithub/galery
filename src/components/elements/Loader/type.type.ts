export type Props = {
  big?: boolean;
  white?: boolean;
};
