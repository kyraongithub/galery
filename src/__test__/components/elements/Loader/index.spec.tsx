import { render } from "@testing-library/react";
import Loader from "@/components/elements/Loader";

describe("loader component", () => {
  it("render loader", () => {
    const { container } = render(<Loader />);
    expect(container).toMatchSnapshot();
  });
});
