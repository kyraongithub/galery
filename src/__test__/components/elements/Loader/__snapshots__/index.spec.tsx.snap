// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loader component render loader 1`] = `
<div>
  <div
    class="loader"
  />
</div>
`;
